### DISCUSSION TITLE
How to use .env.local instead of .env?

### DISCUSSION THREAD

**Discussion by realtebo** [prisma/prisma#25183] [Q&A] [answered]
### Question

I'm using next 14, and we are using .env.local file because nextjs can handle it for local development.

I see that prisma require .env file instead. But we are using .env for something like 'staging' o a pre-demo. In short, we need to keep .env and .env.local differents.

How can we use .env.local with prisma instead of .env?

### How to reproduce (optional)

Nothing to reproduce

**Comment by ludralph** [comment, marked as answer]
Hi @realtebo 👋 

To use `.env.local` with Prisma instead of the default `.env` file, you can use the `dotenv-cli` package to specify which environment file to load. Prisma does not natively support loading `.env.local` files, but you can work around this limitation by using `dotenv-cli`.

1. Install dotenv-cli:
```
npm install dotenv-cli
```

2. Use `dotenv-cli` to load `.env.local`: You can prefix your Prisma commands with `dotenv -e .env.local --` to ensure that the environment variables from `.env.local` are loaded. For example:
```
dotenv -e .env.local -- npx prisma generate
```
This approach allows you to keep your `.env` and `.env.local` files separate and use the appropriate one for your local development with Next.js.

You can also read this section of the [documentation](https://www.prisma.io/docs/orm/more/development-environment/environment-variables/using-multiple-env-files)

**Comment by ghost** [comment]
hey there please help me @ludralph 

i used the command you used with dotenv -e .env.local with the correct env being in .env.local but prisma seems to be loading the env from .env file instead of the .env.local file

```
meqr@mquery:~/Documents/Github/sorti$ npm run generate:dev

> sorti@0.1.0 generate:dev
> dotenv -e .env.local -- npx prisma generate

Environment variables loaded from .env
Prisma schema loaded from prisma/schema.prisma

✔ Generated Prisma Client (v6.2.1) to ./node_modules/@prisma/client in 173ms

Start by importing your Prisma Client (See: https://pris.ly/d/importing-client)

Help us improve the Prisma ORM for everyone. Share your feedback in a short 2-min survey: https://pris.ly/orm/survey/release-5-22
```

**Reply by ludralph** [reply]
Hi @Meqr1 

The output shows that it's been loaded from `.env`, this is misleading.  The .env.local file is actually being read. See this open [issue](https://github.com/prisma/prisma/issues/10104) to address it and this [comment](https://github.com/prisma/prisma/issues/10104#issuecomment-961135194).

**Reply by cheng-dai** [reply]
having the same issue.
